bundles/org.openhab.ui/web/src/components/cards/glance/location/status-badge.vue

```diff
@@ -207,9 +208,29 @@ export default {
       }
     },
     queryLightPoints () {
-      let direct = findPoints(this.element.properties, 'Point', true, 'Property_Light')
-      if (direct.length) return direct
-      return findPoints(allEquipmentPoints(this.element.equipment), 'Point', true, 'Property_Light')
+      // Look for all control points on the location with light property
+      const points = []
+      points.push(...findPoints(this.element.properties, 'Point_Control', true, 'Property_Light'))
+      // Repeat this for equipments on the location, but this time, as it is an equipment, assume it only represents one light and we default to the switch
+      let equipment = findEquipment(this.element.equipment, 'Equipment_LightSource', true)
+      points.push(...this.element.equipment.map((e) => {
+        const isLightSource = equipment.includes(e) // for light source equipment we look beyond property light
+        let equipmentPoints = findPoints(e.points, 'Point_Control_Switch', false, 'Property_Light')
+        if (equipmentPoints.length) return equipmentPoints.slice(0, 1)
+        if (isLightSource) {
+          equipmentPoints = findPoints(e.points, 'Point_Control_Switch', false)
+          if (equipmentPoints.length) return equipmentPoints.slice(0, 1)
+        }
+        equipmentPoints = findPoints(e.points, 'Point_Control', true, 'Property_Light')
+        if (equipmentPoints.length) return equipmentPoints.slice(0, 1)
+        if (isLightSource) {
+          equipmentPoints = findPoints(e.points, 'Point_Control', false)
+          if (equipmentPoints.length) return equipmentPoints.slice(0, 1)
+        }
+        return []
+      }).flat())
+      // If there are no points, use the equipment items themselves
+      return equipment.filter((e) => e.points.length === 0).map((e) => e.item)
```

Comment:
Looking at this again, my last commit (trying to remove the Set duplication removal), also removed the return statement for points. That should go back in.
Also equipment with no points (but an item) will only get added when there are no points at all now.
```suggestion
      points.push(...equipment.filter((e) => e.points.length === 0).map((e) => e.item))
      return points
```